src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java

```diff
@@ -89,53 +89,140 @@
  * <pre>
  * &lt;module name=&quot;RequireThis&quot;/&gt;
  * </pre>
+ * <p>Example:</p>
+ * <pre>
+ * public class Test {
+ *   private int a;
+ *   private int b;
+ *   private int c;
+ *
+ *   public Test(int a) {
+ *     // overlapping by constructor argument
+ *     this.a = a;       // OK, this keyword used
+ *     b = 0;            // OK, no overlap
+ *     foo(5);           // OK
+ *   }
+ *
+ *   public void foo(int c) {
+ *     // overlapping by method argument
+ *     c = c;            // violation, reference to instance variable "c" requires "this"
+ *   }
+ * }
+ * </pre>
  * <p>
- * To configure to check the {@code this} qualifier for fields only:
+ * To configure the check for fields only (with validateOnlyOverlapping set to false):
  * </p>
  * <pre>
  * &lt;module name=&quot;RequireThis&quot;&gt;
  *   &lt;property name=&quot;checkMethods&quot; value=&quot;false&quot;/&gt;
+ *   &lt;property name=&quot;validateOnlyOverlapping&quot; value=&quot;false&quot;/&gt;
```

Comment:
please explain reason of two fields modification in the same time.

it is better to change one field and show validation to let use quickly detect diff when only one parameter of Check is changed.
Each change in config should make some new violation or OK in comparison to default.

Reply:
If I change one property at a time, for example if I change from **default (all properties are true)** to - 
```xml
<module name="RequireThis">
  <property name="checkMethods" value="false"/>
</module>
```

then there will be no difference in number of violations. This is because there is no case where a method is overlapped, so when we turn off checking of methods, **no violation disappears.** This would confuse the user, because it would appear as if nothing changed. So I used it in combination with `validateOnlyOverlapping` to show how both values - **true**/**false** - will affect the given example. 
I did the same combination with both `checkFields` and `checkMethods` to keep the examples uniform and predictable. 
Let me know if I should make any changes.

Reply:
>because there is no case where a method is overlapped, so when we turn off checking of methods, no violation disappears. 

User read doc only when they tried themselves and they do not understand how it works.
So user will be confused before reading. And this doc reading will let them know that it is normal behavior. It is confusing even for me now why option make no effect on validation.

Reply:
Done, I have seperated the examples, and added a line `note that method call does not raise violation because of no overlap`.
Hopefully, it is easier to understand now.

Reply:
much better, thanks a lot !